velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -1060,6 +1062,73 @@ TpchPlan TpchQueryBuilder::getQ10Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ11Plan() const {
+  // Define required columns for each table involved.
+  std::vector<std::string> partsuppColumns = {
+      "ps_partkey", "ps_suppkey", "ps_availqty", "ps_supplycost"};
+  std::vector<std::string> supplierColumns = {"s_suppkey", "s_nationkey"};
+  std::vector<std::string> nationColumns = {"n_nationkey", "n_name"};
+
+  // Define row types for scans.
+  auto partsuppRowType = getRowType(kPartsupp, partsuppColumns);
+  auto supplierRowType = getRowType(kSupplier, supplierColumns);
+  auto nationRowType = getRowType(kNation, nationColumns);
+
+  // Create filters and node IDs.
+  const std::string nationNameFilter = "n_name = 'GERMANY'";
+
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  core::PlanNodeId partsuppScanId, supplierScanId, nationScanId;
+
+  // Scan nation table and filter by name.
+  auto nationScan = PlanBuilder(planNodeIdGenerator)
+                        .tableScan(
+                            kNation,
+                            nationRowType,
+                            getFileColumnNames(kNation),
+                            {nationNameFilter})
+                        .capturePlanNodeId(nationScanId)
+                        .planNode();
+
+  // Join supplier with nation.
+  auto supplierJoinNation =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kSupplier, supplierRowType, getFileColumnNames(kSupplier))
+          .capturePlanNodeId(supplierScanId)
+          .hashJoin(
+              {"s_nationkey"}, {"n_nationkey"}, nationScan, "", {"s_suppkey"})
+          .planNode();
+
+  // Main plan joining partsupp with supplier+nation.
+  auto mainPlan =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kPartsupp, partsuppRowType, getFileColumnNames(kPartsupp))
+          .capturePlanNodeId(partsuppScanId)
+          .hashJoin(
+              {"ps_suppkey"},
+              {"s_suppkey"},
+              supplierJoinNation,
+              "",
+              partsuppColumns)
+          .project(
+              {"ps_supplycost * ps_availqty AS product_cost_qty", "ps_partkey"})
+          .partialAggregation(
+              {"ps_partkey"}, {"sum(product_cost_qty) AS value"})
```

Comment:
This query plan does not seem to account for following clause in query:
```
having
    sum(ps.supplycost * ps.availqty) > (
        select
            sum(ps.supplycost * ps.availqty) * 0.0001
```

Could you please recheck if the plan is correct? You could try to verify correctness with different scale factors.

Reply:
Updated the query plan and verified with scale factor 0.01 and 1.